thriftbp/client_pool.go

```diff
@@ -11,6 +11,7 @@ import (
 
 	"github.com/apache/thrift/lib/go/thrift"
 	"github.com/avast/retry-go"
+	lru "github.com/hashicorp/golang-lru/v2"
```

Comment:
In the past, we've been hesitant to use this library because it included patented code:

https://github.com/hashicorp/golang-lru/blob/56a2dc02f65b563d946d7ce80b24f905f3519d84/arc/arc.go#L18

But it looks like they moved that into it's own package in v2.0.3

https://github.com/hashicorp/golang-lru/releases/tag/v2.0.3

so maybe it's ok now?